skin/src/main/java/org/researchstack/skin/ui/fragment/ActivitiesFragment.java

```diff
@@ -87,30 +110,99 @@ private void setUpAdapter() {
                 .compose(ObservableUtils.applyDefault())
                 .map(o -> (SchedulesAndTasksModel) o)
                 .subscribe(model -> {
-                    adapter = new TaskAdapter(model);
-                    recyclerView.setAdapter(adapter);
+                    swipeContainer.setRefreshing(false);
+                    if(adapter == null) {
+                        adapter = new TaskAdapter(getActivity());
+                        recyclerView.setAdapter(adapter);
 
-                    subscription = adapter.getPublishSubject().subscribe(task -> {
+                        subscription = adapter.getPublishSubject().subscribe(task -> {
 
-                        Task newTask = DataProvider.getInstance().loadTask(getContext(), task);
+                            Task newTask = DataProvider.getInstance().loadTask(getContext(), task);
 
-                        if (newTask == null) {
-                            Toast.makeText(getActivity(),
-                                    R.string.rss_local_error_load_task,
-                                    Toast.LENGTH_SHORT).show();
-                            return;
-                        }
+                            if (newTask == null) {
+                                Toast.makeText(getActivity(),
+                                        R.string.rss_local_error_load_task,
+                                        Toast.LENGTH_SHORT).show();
+                                return;
+                            }
+
+                            startActivityForResult(ViewTaskActivity.newIntent(getContext(), newTask),
+                                    REQUEST_TASK);
+                        });
+                    } else {
+                        adapter.clear();
+                    }
+
+                    adapter.addAll(processResults(model));
 
-                        startActivityForResult(ViewTaskActivity.newIntent(getContext(), newTask),
-                                REQUEST_TASK);
-                    });
                 });
     }
 
+    /**
+     * Process the model to create section groups and section headers
+     * @param model
+     * @return
+     */
+    private List<Object> processResults(SchedulesAndTasksModel model) {
+        List<Object> tasks = new ArrayList<>();
+
+        DateTime now = new DateTime();
+        DateTime startOfDay = new DateTime().withTimeAtStartOfDay().minusSeconds(1);
```

Comment:
Its a bit hacky I agree.  Some of the activites are scheduled on the exact start of the day so the isBefore() and isAfter() methods were not working since it is equal.  This was a quick and dirty solution.